### DISCUSSION TITLE
Failing to match experimental pharmacophore data with EmbedLib.MatchPharmacophoreToMol

### DISCUSSION THREAD

**Discussion by OlgaGKononova** [rdkit/rdkit#6661] [Q&A]
Hi there!

I am going over the ePharmaLib dataset (https://pubs.acs.org/doi/10.1021/acs.jcim.1c00135) that contains >1K crystal structures references (PDB IDs) and corresponding pharmacophoric points extracted based on the crystalized ligand. I am following procedure from [this notebook](Notebooks/Stiefl_RDKitPh4FullPublication.ipynb). However, when I run `EmbedLib.MatchPharmacophoreToMol` on pharmacophore and ligand from these experimental data, the result is that molecule doesn't match pharmacophore. 

Specifically, procedure is following:

Example file from the dataset looks like this:
```
1a27-EST-DHB1_HUMAN
AROM	13.0545	8.4995	-9.9134	0.7	0	0	0	0
LIPO	13.3484	6.0985	-9.3487	0.7	0	0	0	0
LIPO	13.1685	3.2109	-6.3851	0.7	0	0	0	0
HDON	12.6708	1.9111	-3.7609	1.0	0	0	0	0
EXCL	10.736000061	3.52300000191	-0.600000023842	1.4	0	0.000	0.000	0.000
EXCL	11.0059995651	3.72600007057	-1.98099994659	1.4	0	0.000	0.000	0.000
...
```
I parse it to extract information about pharmacophoric features and construct RDKit pharmacophore:
```
exp_pharm = [
    {'feature_type': 'Aromatic', 'coords': [13.0545, 8.4995, -9.9134]},
    {'feature_type': 'LumpedHydrophobe', 'coords': [13.3484, 6.0985, -9.3487]},
    {'feature_type': 'LumpedHydrophobe', 'coords': [13.1685, 3.2109, -6.3851]},
    {'feature_type': 'Donor', 'coords': [12.6708, 1.9111, -3.7609]}
]
```
and convert into RDKit pharmacophore:
```
def get_pharmacophore(parsed_data):
    rdk_pharm = [ChemicalFeatures.FreeChemicalFeature(d["feature_type"], Geometry.Point3D(*d["coords"]))
                for d in parsed_data]
    rdk_pharm = Pharmacophore.Pharmacophore(rdk_pharm)
    return rdk_pharm
```
```
rdk_pharm = get_pharmacophore(exp_pharm)
```
Resulted RDKit pharmacophore:
```
Aromatic [13.0545, 8.4995, -9.9134]
LumpedHydrophobe [13.3484, 6.0985, -9.3487]
LumpedHydrophobe [13.1685, 3.2109, -6.3851]
Donor [12.6708, 1.9111, -3.7609]
```

Then I get PDB structure `1a27` and extract ligand from there. Technically, it has 2 ligands within pocket, but only one ligand overlaps well with the features provided in the file. 
![image](https://github.com/rdkit/rdkit/assets/11651311/c67c4f4e-88b0-4fee-874f-2fe65068b597)

I take that ligand from PDB, and run matching with `EmbedLib.MatchPharmacophoreToMol`
```
features_path = "./BaseFeatures.fdef"
featFactory = ChemicalFeatures.BuildFeatureFactory(features_path)

ligand = Chem.MolFromPDBBlock(ligand_pdb_block)
for feat in featFactory.GetFeaturesForMol(ligand):
    print(feat.GetFamily(), list(feat.GetPos()))
```
Printing result:
```
Donor [12.024, 10.68, -11.119]
Donor [12.842, 2.32, -4.004]
Acceptor [12.024, 10.68, -11.119]
Acceptor [12.842, 2.32, -4.004]
Hydrophobe [13.017, 7.352, -9.897]
Hydrophobe [13.048, 5.027, -8.168]
Hydrophobe [13.444, 6.405, -7.591]
Hydrophobe [13.039, 7.545, -8.496]
Hydrophobe [12.906, 4.078, -5.773]
Hydrophobe [13.311, 3.906, -7.241]
LumpedHydrophobe [13.259166666666665, 6.202166666666667, -9.035999999999998]
LumpedHydrophobe [13.212499999999999, 5.230499999999999, -6.6884999999999994]
```
However, `EmbedLib.MatchPharmacophoreToMol(ligand, featFactory, rdk_pharm)` returns `False`. 

I tried to manually construct list of matches for pharmacophore features based on proximity of ligand features to pharmacophore features and feed it to `EmbedLib.MatchPharmacophore`. Everything runs correctly, but returns failing as output.

I will appreciate if you can explain to me why `EmbedLib.MatchPharmacophoreToMol` cannot identify a match here. I ran this procedure also for several other data in the ePharmaLib dataset - cannot match neither of them although the pharmacophore features nicely overlap with the ligand.

**Comment by bp-kelley** [comment]
Can you print the smiles of the ligand?   Is it being processed correctly?  If protonation is wrong, I can see bad aromaticity and/or acceptor-donor.
